include/odp/api/spec/ipsec.h

```diff
@@ -1189,7 +1189,13 @@ typedef struct odp_ipsec_out_inline_param_t {
 	struct {
 		/** Points to first byte of outer headers to be copied in
 		 *  front of the outgoing IPSEC packet. Implementation copies
-		 *  the headers during odp_ipsec_out_inline() call. */
+		 *  the headers during odp_ipsec_out_inline() call.
+		 *
+		 *  Null value indicates that the outer headers are in the
+		 *  packet data, starting at L2 offset and ending at the byte
+		 *  before L3 offset. In this case L2 offset must be set and
+		 *  less than L3 offset and field 'len' will be ignored.
```

Comment:
After thinking this again with Janne, it could be better to require that application sets len also when ptr ==NULL. This way we have backwards compatible option for "no outer header defined" case (ptr==NULL && len==0) and also later we can allow shorter outer headers than l3-l2_offset. Anyway, we could start with the strict requirement that outer header is just before l3_offset. So, the last sentence could be modified something like this:

Null value indicates that the outer headers are in the
packet data, starting at L2 offset and ending at the byte
before L3 offset. In this case, value of 'len' field must be 
greater than zero and set to L3 offset minus L2 offset.

Reply:
Agreed. Unrelated to the PR, one minor clarification. What should be the effect of 

"no outer header defined" case (ptr==NULL && len==0)

Should that be reported as an error?

Reply:
That would be against the API spec so an application is not allowed to make such call. If it does, the behavior is undefined. In addition to the popular choice of starting nethack, the implementation could e.g. process the packet as if no header prepending was requested, but an application must not rely on that (ptr!=NULL, len==0 would be the only API-defined way to request zero-lenght header).